src/build.d

```diff
@@ -0,0 +1,1143 @@
+#!/usr/bin/env rdmd
+/**
+DMD builder
+
+Usage:
+  ./build.d dmd
+
+TODO:
+- build.d should be a default dependency to all rules
+- improve rebuild dependencies (e.g. header files should trigger a rebuild)
+- add different ENABLE modes
+- add all posix.mak Makefile targets
+- support 32-bit builds
+- test on OSX
+- test on Windows
+- allow appending DFLAGS via the environment
+- evaluate whether to use globbing for the file listings
```

Comment:
I know it's a long list, but imho it's better to have something as an initial base on which incremental reviews can be done.
Also with an existing build.d other people can more easily join this effort.